homeassistant/components/august/binary_sensor.py

```diff
@@ -162,12 +177,31 @@ def device_info(self):
             "model": self._model,
         }
 
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+
+        @callback
+        def update():
+            """Update the state."""
+            self.async_schedule_update_ha_state(True)
```

Comment:
Same thing as I commented earlier, we should use `self.async_write_ha_state()` here.

Reply:
Right now I'm stuck doing the update because it needs fetch the device detail and then update it with the activity that just came in.  Since fetching the detail can do I/O I don't think I can do that here.

If I do the refactor I above to add detail fetch methods I could add a method to consume the pending activity without the chance I/O happens and then write the state.

Let me know if you want me to do the refactor in this PR or a followup.

Reply:
Follow-up is fine.

Reply:
@balloob  done in https://github.com/home-assistant/home-assistant/pull/32249